CalorieTracker/CalorieTracker/CalorieTrackerViewController.swift

```diff
@@ -0,0 +1,188 @@
+        super.viewWillAppear(true)
+        self.viewDidLoad()
+        tableView.reloadData()
+
+    }
+    @objc func updateViews() {
+        let chart = Chart(frame: view.frame)
+        guard let sections = fetchedResultsController.sections else { return }
+        chart.removeAllSeries()
+        var seriesOfDoubles: [Double] = []
+        for section in sections {
+            if let objects = section.objects as? [Calorie] {
+                for object in objects {
+                    seriesOfDoubles.append(Double(object.amount))
+                }
+            }
+        }
+        DispatchQueue.main.async {
+            self.chart.add(ChartSeries(seriesOfDoubles))
+            self.tableView.reloadData()
+        }
+    }
+    //MARK: - IBACTION -
+    @IBAction func addButtonTapped(_ sender: Any) {
+        // Create alert
+        let alert = UIAlertController(title: "Add calorie intake",
+                                      message: "Enter the amount of calories",
+                                      preferredStyle: .alert)
+
+        // Add text field
+        alert.addTextField { textField in
+            textField.placeholder = "Calories"
+        }
+        alert.addAction(UIAlertAction(title: "Save", style: .default, handler: { _ in
+            if alert.textFields?.first?.text?.isNumeric == false {
+                self.didNotEnterCaloriesAlert()
+                print("Empty")
+            }
+            guard let currentTextField = alert.textFields,
+                let caloriesString = currentTextField[0].text,
+                let calories = Int(caloriesString)
+                else { return }
+            // Create calories from given integer.
+            self.calorieController.add(amount: calories)
+        }))
+        alert.addAction(UIAlertAction(title: "Cancel",
+                                      style: .destructive,
+                                      handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+    func didNotEnterCaloriesAlert() {
+        let alert = UIAlertController(title: "Error Adding Calories",
+                                      message: "Need to enter a number.",
+                                      preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "OK",
+                                      style: .default,
+                                      handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+}
+
+extension CalorieTrackerViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle,forRowAt indexPath:IndexPath) {
+        if editingStyle == .delete {
+            let calorie = fetchedResultsController.object(at: indexPath)
+            calorieController.delete(calorie: calorie) { _ in
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
```

Comment:
You don't need to manually reload the table view since you are using a FRC

``` swift
func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle,forRowAt indexPath:IndexPath) {
    if editingStyle == .delete {
        let calorie = fetchedResultsController.object(at: indexPath)
        calorieController.delete(calorie: calorie)
    }
}
```